pkg/components/discovery.go

```diff
@@ -23,7 +24,10 @@ func NewDiscovery(cfgen *ytconfig.Generator, ytsaurus *apiproxy.Ytsaurus) *Disco
 		APIProxy:       ytsaurus.APIProxy(),
 		ComponentLabel: consts.YTComponentLabelDiscovery,
 		ComponentName:  "Discovery",
-		MonitoringPort: consts.DiscoveryMonitoringPort,
+	}
+
+	if resource.Spec.Discovery.InstanceSpec.MonitoringPort == nil {
+		resource.Spec.Discovery.InstanceSpec.MonitoringPort = ptr.Int32(consts.DiscoveryMonitoringPort)
```

Comment:
IMHO changing spec isn't a good idea.

Reply:
I've suggested to make make a function, which would update all instanceSpecs in ytsaurusSpec, for start it can
  - sets monitoring port to default if not specified
  - sets image to coreImage if not specified
This function should be called early in the reconcillation loop.
That way all our code can expect actual values in InstanceSpec and get rid of ifs.

For a cleaner solution here may be introduced some new struct, in which we will put values from spec/instanceSpec with beforementined logic and use this struct, not the original instanceSpec later. But i suppose this would be rather big change, not sure if it worth it.

Reply:
well, external data mutation by pointer is not the best approach for sure. what you can propose here? would it be better to accept a struct rather than a pointer?